CRM/Price/Page/Field.php

```diff
@@ -230,8 +230,13 @@ public function run() {
     );
 
     if ($this->_sid) {
+      $usedByDefaults = [
+        'civicrm_event' => FALSE,
+        'civicrm_event' => FALSE,
+        'civicrm_contribution_page' => FALSE,
+      ];
       $usedBy = CRM_Price_BAO_PriceSet::getUsedBy($this->_sid);
-      $this->assign('usedBy', $usedBy);
+      $this->assign('usedBy', array_merge($usedByDefaults, $usedBy));
```

Comment:
@eileenmcnaughton @braders this line caused a regression. The $usedBy will always have 
```php
$usedByDefaults = [
        'civicrm_event' => FALSE,
        'civicrm_contribution_page' => FALSE,
      ];
 ```
 irrespective of $usedBy value, even if it's empty. 
As a result, it will prevent any price field to get deleted.  Because of this condition in Field.tpl
```
{elseif ($usedBy and $action eq 8) or $usedBy.civicrm_event or $usedBy.civicrm_contribution_page}
  <div id="price_set_used_by" class="messages status no-popup">
    {icon icon="fa-info-circle"}{/icon}
    {if $action eq 8}
      {ts 1=$usedPriceSetTitle}Unable to delete the '%1' Price Field - it is currently in use by one or more active events or contribution pages or contributions  or event templates.{/ts}
    {/if}
```

The fix would be 
```diff
diff --git a/CRM/Price/Page/Field.php b/CRM/Price/Page/Field.php
index 72ab8d481a..c1a70040fd 100644
--- a/CRM/Price/Page/Field.php
+++ b/CRM/Price/Page/Field.php
@@ -231,12 +231,11 @@ class CRM_Price_Page_Field extends CRM_Core_Page {
 
     if ($this->_sid) {
       $usedByDefaults = [
-        'civicrm_event' => FALSE,
         'civicrm_event' => FALSE,
         'civicrm_contribution_page' => FALSE,
       ];
       $usedBy = CRM_Price_BAO_PriceSet::getUsedBy($this->_sid);
-      $this->assign('usedBy', array_merge($usedByDefaults, $usedBy));
+      $this->assign('usedBy', array_intersect_key($usedByDefaults, $usedBy));
       $this->_isSetReserved = CRM_Core_DAO::getFieldValue('CRM_Price_DAO_PriceSet', $this->_sid, 'is_reserved');
       $this->assign('isReserved', $this->_isSetReserved);
```

Reply:
thanks @monishdeb  - are you going to PR in the fix to the rc?

Reply:
Yep, created a ticket https://lab.civicrm.org/dev/core/-/issues/3999 to refer the rc PR. I'll let you know after submitting the PR.

Reply:
thanks!